build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/testfixtures/TestFixturesPlugin.java

```diff
@@ -0,0 +1,231 @@
+    }
+
+    private void maybeSkipTask(Provider<DockerSupportService> dockerSupport, TaskProvider<Task> task) {
+        task.configure(t -> maybeSkipTask(dockerSupport, t));
+    }
+
+    private void maybeSkipTask(Provider<DockerSupportService> dockerSupport, Task task) {
+        task.onlyIf(spec -> {
+            boolean isComposeAvailable = dockerSupport.get().getDockerAvailability().isComposeAvailable;
+            if (isComposeAvailable == false) {
+                LOGGER.info("Task {} requires docker-compose but it is unavailable. Task will be skipped.", task.getPath());
+            }
+            return isComposeAvailable;
+        });
+    }
+
+    private void configureServiceInfoForTask(
+        Task task,
+        Project fixtureProject,
+        boolean enableFilter,
+        BiConsumer<String, Integer> consumer
+    ) {
+        // Configure ports for the tests as system properties.
+        // We only know these at execution time so we need to do it in doFirst
+        task.doFirst(new Action<Task>() {
+            @Override
+            public void execute(Task theTask) {
+                TestFixtureExtension extension = theTask.getProject().getExtensions().getByType(TestFixtureExtension.class);
+
+                fixtureProject.getExtensions()
+                    .getByType(ComposeExtension.class)
+                    .getServicesInfos()
+                    .entrySet()
+                    .stream()
+                    .filter(entry -> enableFilter == false || extension.isServiceRequired(entry.getKey(), fixtureProject.getPath()))
+                    .forEach(entry -> {
+                        String service = entry.getKey();
+                        ServiceInfo infos = entry.getValue();
+                        infos.getTcpPorts().forEach((container, host) -> {
+                            String name = "test.fixtures." + service + ".tcp." + container;
+                            theTask.getLogger().info("port mapping property: {}={}", name, host);
+                            consumer.accept(name, host);
+                        });
+                        infos.getUdpPorts().forEach((container, host) -> {
+                            String name = "test.fixtures." + service + ".udp." + container;
+                            theTask.getLogger().info("port mapping property: {}={}", name, host);
+                            consumer.accept(name, host);
+                        });
+                    });
+            }
+        });
+    }
+
+    @SuppressWarnings("unchecked")
+    private Class<? extends DefaultTask> getTaskClass(String type) {
+        Class<?> aClass;
+        try {
+            aClass = Class.forName(type);
+            if (DefaultTask.class.isAssignableFrom(aClass) == false) {
+                throw new IllegalArgumentException("Not a task type: " + type);
+            }
+        } catch (ClassNotFoundException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)